crypto/common/primes.go

```diff
@@ -39,6 +39,26 @@ func GetSafePrime(bits int) (p *big.Int, err error) {
 	}
 }
 
+// GetSpecialRSAPrimes returns primes P, Q, p, q such that P = 2*p + 1 and Q = 2*q + 1.
+func GetSpecialRSAPrimes(bits int) (P, Q, p, q *big.Int, err error) {
```

Comment:
For better readability, we could maybe make a `struct SpecialRSAPrimes` with fields `P,Q,p,q`, and return this struct here. In my opinion, it's better to return two values from a function (pointer to a struct and error) instead of 4 `*big.Int`s and error :slightly_smiling_face: At least from perspective of the caller, who has to deal with these returned values (because he has to be aware of the order in which these values are returned...)